File: Reverse and add/revadd.java
import java.util.Scanner;

public class ReverseAndAdd {

    static boolean isPalindrome(int num) {
        String numStr = Integer.toString(num);
        return numStr.equals(new StringBuilder(numStr).reverse().toString());
    }

    static int[] reverseAndAdd(int n) {
        int times = 0;
        while (!isPalindrome(n)) {
            n += Integer.parseInt(new StringBuilder(Integer.toString(n)).reverse().toString());
            times++;
        }
        return new int[]{n, times};
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter n: ");
        int n = sc.nextInt();

        int[] result = reverseAndAdd(n);
        System.out.println(result[1] + " " + result[0]);

        sc.close();
    }
}

File: Reverse and add/revadd.py
def reverse_and_add(n):
    def is_palindrome(num):
        return str(num) == str(num)[::-1]

    times = 0
    while not is_palindrome(n):
        n += int(str(n)[::-1])
        times += 1

    return n, times

n = int(input("Enter n: "))
result, times = reverse_and_add(n)
print(times, result)